plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -423,6 +424,44 @@ public static PinotColumnHandle toNonAggregateColumnHandle(PinotColumnHandle col
         return new PinotColumnHandle(columnHandle.getColumnName(), columnHandle.getDataType(), quoteIdentifier(columnHandle.getColumnName()), false, false, true, Optional.empty(), Optional.empty());
     }
 
+    private boolean canPushDownAggregateIfGroupingColumnsPresent(ConnectorSession session, Optional<DynamicTable> dynamicTable, List<AggregateFunction> aggregates, Map<String, ColumnHandle> assignments)
```

Comment:
it looks like this method does things that do not match what the name suggests (see pattern matching `implementCountDistinct.getPattern().matches(`)

Reply:
lmk if the new name is ok: `isAggregationPushdownSupported`. Thanks for all the advice!